client.go

```diff
@@ -87,39 +90,69 @@ func (c *Client) SetID(id string) error {
 	return nil
 }
 
-func (c *Client) Consumer(topics []string, cfg rdkafka.ConfigMap) (*Consumer, error) {
-	if !c.consReady {
-		return nil, errors.New("Connection not ready. Identify yourself using `CLIENT SETNAME` first")
-	}
+// registerConsumer registers a new Consumer denoted by cid.
+func (c *Client) registerConsumer(
+	cid ConsumerID, gid string, topics []string, cfg rdkafka.ConfigMap) (*Consumer, error) {
```

Comment:
As a sanity check, I think we should assert that gid is not an empty string. If it is, we should return an error.

Reply:
The `gid` attribute is set by the `SetID` function and there's no code-path to call `registerConsumer` without calling the `SetID` function first.

However, we could add an extra layer of protection here.